### PR TITLE
add a flag to control whether enable kubeconfig keychain

### PR DESCRIPTION
**Pull request by changweige**
If node has no kubelet deployed, listener will report warning forver:

E0815 10:37:46.330502 1446182 reflector.go:138] pkg/mod/k8s.io/client-go@v0.22.5/tools/cache/reflector.go:167: Failed to watch *v1.Secret: failed to list *v1.Secret: Get "https://127.0.0.1:27949/api/v1/secrets?fieldSelector=type%3Dkubernetes.io%2Fdockerconfigjson&limit=500&resourceVersion=0": dial tcp 127.0.0.1:27949: connect: connection refused
E0815 10:38:30.398522 1446182 reflector.go:138] pkg/mod/k8s.io/client-go@v0.22.5/tools/cache/reflector.go:167: Failed to watch *v1.Secret: failed to list *v1.Secret: Get "https://127.0.0.1:27949/api/v1/secrets?fieldSelector=type%3Dkubernetes.io%2Fdockerconfigjson&limit=500&resourceVersion=0": dial tcp 127.0.0.1:27949: connect: connection refused
E0815 10:39:08.890713 1446182 reflector.go:138] pkg/mod/k8s.io/client-go@v0.22.5/tools/cache/reflector.go:167: Failed to watch *v1.Secret: failed to list *v1.Secret: Get "https://127.0.0.1:27949/api/v1/secrets?fieldSelector=type%3Dkubernetes.io%2Fdockerconfigjson&limit=500&resourceVersion=0": dial tcp 127.0.0.1:27949: connect: connection refused

Signed-off-by: Changwei Ge <gechangwei@bytedance.com>

### CODE REVIEW

File: cmd/containerd-nydus-grpc/pkg/command/flags.go [hunk 1]
```diff
@@ -209,6 +210,12 @@ func buildFlags(args *Args) []cli.Flag {
 			Usage:       "path to the kubeconfig file",
 			Destination: &args.KubeconfigPath,
 		},
+		&cli.BoolFlag{
+			Name:        "enable-kubeconfig-keychain",
+			Value:       false,
+			Usage:       "Listen on K8s API server to synchronize dockerconfigjson",
```
Comment (imeoer) [inline cmd/containerd-nydus-grpc/pkg/command/flags.go:216]: May be:

```
synchronize `kubernetes.io/dockerconfigjson` secret from kubernetes API server with provided `--kubeconfig-path` (default `$KUBECONFIG` or `~/.kube/config`).
```

File: pkg/auth/kubesecret.go [hunk 2]
```diff
@@ -43,10 +43,10 @@ func InitKubeSecretListener(ctx context.Context, kubeconfigPath string) {
 		if kubeconfigPath != "" {
 			_, err := os.Stat(kubeconfigPath)
 			if err != nil && !os.IsNotExist(err) {
-				logrus.WithError(err).Warningf("kubeconfig does not exist, kubeconfigPath %s", kubeconfigPath)
+				logrus.WithError(err).Errorf("kubeconfig does not exist, kubeconfigPath %s", kubeconfigPath)
```
Comment (imeoer) [inline pkg/auth/kubesecret.go:46]: Warn level is still ok?

Comment (changweige) [reply pkg/auth/kubesecret.go]: fixed

File: cmd/containerd-nydus-grpc/pkg/command/flags.go (hunk 1 above)
Comment (changweige) [reply cmd/containerd-nydus-grpc/pkg/command/flags.go]: fixed

**Review by imeoer — APPROVED** [review]

**Pull request merged**